graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/validation/validateFieldSet.kt

```diff
@@ -32,8 +31,7 @@ internal fun validateKeySetField(targetField: GraphQLFieldDefinition?, extendedT
             errors.add("$validatedDirective specifies invalid field set - type incorrectly references external field=${targetField.name}")
         }
 
-        when (GraphQLTypeUtil.unwrapNonNull(targetField.type)) {
-            is GraphQLList -> errors.add("$validatedDirective specifies invalid field set - field set references GraphQLList, field=${targetField.name}")
+        when (targetField.type.unwrapType()) {
             is GraphQLInterfaceType -> errors.add("$validatedDirective specifies invalid field set - field set references GraphQLInterfaceType, field=${targetField.name}")
```

Comment:
Question: Do we need to check all these other types too? If it can just be on any `FIELD_DEFINITION` why do we need to exclude interfaces and unions?